src/main/java/org/terasology/minimap/rendering/nui/layers/MinimapGrid.java

```diff
@@ -283,27 +284,36 @@ private void renderDirtyBlock(Canvas canvas, int chunkX, int chunkZ, Vector3i po
     }
 
     private void drawPlayerArrows(Canvas canvas, float zoom, int centerX, int centerZ) {
-        // draw arrowhead
-        Texture arrowhead = Assets.getTexture("Minimap:arrowhead").get();
-        // Drawing textures with rotation is not yet supported, see #1926
-        // We therefore use a workaround based on mesh drawing
-        // The width of the screenArea is doubled to avoid clipping issues when the texture is rotated
+
         int width = getPreferredContentSize().getX();
         int height = getPreferredContentSize().getY();
-        int arrowWidth = arrowhead.getWidth() * 2;
-        int arrowHeight = arrowhead.getHeight() * 2;
-        int arrowX = (width - arrowWidth) / 2;
-        int arrowY = (height - arrowHeight) / 2;
-//        canvas.drawTexture(arrowhead, arrowX, arrowY, rotation);
 
-        // UITexture should be used here, but it doesn't work
-        Material material = Assets.getMaterial("engine:UILitMesh").get();
-        material.setTexture("texture", arrowhead);
-        Mesh mesh = Assets.getMesh("engine:UIBillboard").get();
 
         for (EntityRef alivePlayer : alivePlayers) {
             LocationComponent playerLocationComponent = alivePlayer.getComponent(LocationComponent.class);
-            if (playerLocationComponent != null) {
+            MinimapIconComponent minimapIconComponent = alivePlayer.getComponent(MinimapIconComponent.class);
+            if (playerLocationComponent != null && minimapIconComponent != null) {
+                // draw arrowhead
+                if (!Assets.getTexture(minimapIconComponent.value).isPresent()) {
+                    continue;
+                }
+
+                Texture arrowhead = Assets.getTexture(minimapIconComponent.value).get();
+                // Drawing textures with rotation is not yet supported, see #1926
+                // We therefore use a workaround based on mesh drawing
+                // The width of the screenArea is doubled to avoid clipping issues when the texture is rotated
+                int arrowWidth = arrowhead.getWidth() * 2;
+                int arrowHeight = arrowhead.getHeight() * 2;
+                int arrowX = (width - arrowWidth) / 2;
+                int arrowY = (height - arrowHeight) / 2;
+                //canvas.drawTexture(arrowhead, arrowX, arrowY, rotation);
+
+                // UITexture should be used here, but it doesn't work
+                Material material = Assets.getMaterial("engine:UILitMesh").get();
+                material.setTexture("texture", arrowhead);
```

Comment:
https://github.com/Terasology/LightAndShadowResources/pull/31
This wasn't working with atlas.

Reply:
Does it throw an error, or is it _just not working_? In that case, please update the docs accordingly and create an issue - I don't think that's the intended behavior.